src/enforester.js

```diff
@@ -1313,26 +1337,23 @@ export class Enforester {
 
   enforestNewExpression() {
     this.matchKeyword('new');
-    let callee;
-    if (this.isKeyword(this.peek(), 'new')) {
-      callee = this.enforestNewExpression();
-    } else if (this.isKeyword(this.peek(), 'super')) {
-      callee = this.enforestExpressionLoop();
-    } else if (this.isPunctuator(this.peek(), '.') && this.isIdentifier(this.peek(1), 'target')) {
+    let lookahead = this.peek();
+    if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
       this.advance();
       this.advance();
       return new Term('NewTargetExpression', {});
     } else {
-      callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
```

Comment:
I don't think you need to assign to `this.term` here, the local `callee` should work better.